Simplenote/CoreDataManager.swift

```diff
@@ -9,27 +15,48 @@ class CoreDataManager: NSObject {
     private(set) var managedObjectContext: NSManagedObjectContext!
     private(set) var persistentStoreCoordinator: NSPersistentStoreCoordinator!
 
-    init(_ storageURL: URL, storageSettings: StorageSettings = StorageSettings()) throws {
+    init(_ storageURL: URL, storageSettings: StorageSettings = StorageSettings(), for usageType: CoreDataUsageType = .standard) throws {
         super.init()
+        managedObjectModel = NSManagedObjectModel(contentsOf: storageSettings.modelURL)!
+        managedObjectContext = buildMainContext()
+        persistentStoreCoordinator = try buildStoreCoordinator(with: managedObjectModel, storageURL: storageURL)
 
-        guard let mom = NSManagedObjectModel(contentsOf: storageSettings.modelURL) else {
-            fatalError("Cannot load model")
-        }
-        managedObjectModel = mom
+        setupStackIfNeeded(mainContext: managedObjectContext, psc: persistentStoreCoordinator, for: usageType)
+    }
 
-        managedObjectContext = NSManagedObjectContext(concurrencyType: .mainQueueConcurrencyType)
-        managedObjectContext.undoManager = nil
+    private func buildMainContext() -> NSManagedObjectContext {
+        let moc = NSManagedObjectContext(concurrencyType: .mainQueueConcurrencyType)
+        moc.undoManager = nil
+        return moc
+    }
 
+    private func buildStoreCoordinator(with model: NSManagedObjectModel, storageURL: URL) throws -> NSPersistentStoreCoordinator {
         let psc = NSPersistentStoreCoordinator(managedObjectModel: managedObjectModel)
         let options = [NSMigratePersistentStoresAutomaticallyOption: true,
                        NSInferMappingModelAutomaticallyOption: true]
 
         NSLog("🎯 Loading PersistentStore at URL: \(storageURL)")
-        try psc.addPersistentStore(ofType: NSSQLiteStoreType,
+        do {
+            try psc.addPersistentStore(ofType: NSSQLiteStoreType,
                                        configurationName: nil,
                                        at: storageURL,
                                        options: options)
+        } catch {
+            throw NSError(domain: Constants.errorDomain, code: .zero, userInfo: nil)
+        }
+        return psc
+    }
 
-        persistentStoreCoordinator = psc
+    private func setupStackIfNeeded(mainContext: NSManagedObjectContext, psc: NSPersistentStoreCoordinator, for usageType: CoreDataUsageType) {
+        switch usageType {
+        case .standard:
+            break
+        case .intents, .widgets:
+            managedObjectContext.persistentStoreCoordinator = persistentStoreCoordinator
+        }
```

Comment:
In the main app Simperium makes the connections between the different pieces of core data, but in the widgets or intents we don't spin up Simperium so the PSC and the MOC need to be connected so that they can actually work.

This change method is there to check if we are in the main app, and setup the core data based on the usage type

Reply:
Oh I see, thank you for the explanation!